addons/mail/static/src/components/discuss/tests/discuss_tests.js

```diff
@@ -2893,6 +2894,186 @@ QUnit.test('post a simple message', async function (assert) {
     );
 });
 
+QUnit.test('post message on non-mailing channel with "Enter" keyboard shortcut', async function (assert) {
+    assert.expect(2);
+
+    // channel expected to be found in the sidebar
+    // with a random unique id that will be referenced in the test
+    this.data['mail.channel'].records.push({ id: 20, mass_mailing: false });
+    await this.start({
+        discuss: {
+            params: {
+                default_active_id: 'mail.channel_20',
+            },
+        },
+        async mockRPC(route, args) {
+            const res = await this._super(...arguments);
+            if (args.method === 'message_post') {
+                assert.step('message_post');
+            }
+            return res;
+        },
+    });
+
+    // insert some HTML in editable
+    await afterNextRender(() => {
+        document.querySelector(`.o_ComposerTextInput_textarea`).focus();
+        document.execCommand('insertText', false, "Test");
+    });
+    await afterNextRender(() => {
+        const kevt = new window.KeyboardEvent('keydown', { key: "Enter" });
+        document.querySelector('.o_ComposerTextInput_textarea').dispatchEvent(kevt);
+    });
+    assert.verifySteps(
+        ['message_post'],
+        "should have posted message on pressing 'Enter' in text input of composer"
+    );
```

Comment:
In non-"composer component only" tests, I think we can directly assert the new message is on the message list instead of the rpc.